src/GitTools.Core/Context/AuthenticationContext.cs

```diff
@@ -0,0 +1,24 @@
+﻿namespace GitTools
+{
+    public class AuthenticationContext : IAuthenticationContext
+    {
+        public AuthenticationContext()
+        {
+        }
+
+        public AuthenticationContext(string username, string password)
+        {
+            Username = username;
+            Password = password;
+        }
+
+        public AuthenticationContext(string token)
+        {
+            Token = token;
+        }
+
+        public string Username { get; set; }
+        public string Password { get; set; }
```

Comment:
I have considered that, but we pass everything around as strings (into the command line). Do you think it will be better? It's an easy change.